src/client.ts

```diff
@@ -16,109 +31,187 @@ export type ResourceIteratee<T> = (each: T) => Promise<void> | void;
+      throw toThrow;
+    }
+    if (!response.data || !response.data.length) {
+      const toThrow = new IntegrationError({
+        message: 'response is missing data',
+        code: 'API_CLIENT_RESPONSE_VERIFICATION_ERROR',
+      });
+      this.logger.error(toThrow, response);
+      throw toThrow;
+    }
+  }
 
   public async verifyAuthentication(): Promise<void> {
-    // TODO make the most light-weight request possible to validate
-    // authentication works with the provided credentials, throw an err if
-    // authentication fails
-    const request = new Promise<void>((resolve, reject) => {
-      http.get(
-        {
-          hostname: 'localhost',
-          port: 443,
-          path: '/api/v1/some/endpoint?limit=1',
-          agent: false,
-          timeout: 10,
-        },
-        (res) => {
-          if (res.statusCode !== 200) {
-            reject(new Error('Provider authentication failed'));
-          } else {
-            resolve();
-          }
-        },
-      );
+    const uri = '/api/account/get-account';
+    const request = got.post(BASE_URI + uri, {
+      headers: this.generateHeaders(uri),
+      body: EMPTY_BODY,
     });
 
     try {
-      await request;
+      const result = await request;
+      this.verifyResponse(JSON.parse(result.body) as ApiResponse<Account>);
+      console.log(result);
     } catch (err) {
       throw new IntegrationProviderAuthenticationError({
         cause: err,
-        endpoint: 'https://localhost/api/v1/some/endpoint?limit=1',
+        endpoint: BASE_URI + uri,
         status: err.status,
         statusText: err.statusText,
       });
     }
   }
 
-  /**
-   * Iterates each user resource in the provider.
-   *
-   * @param iteratee receives each resource to produce entities/relationships
-   */
-  public async iterateUsers(
-    iteratee: ResourceIteratee<AcmeUser>,
-  ): Promise<void> {
-    // TODO paginate an endpoint, invoke the iteratee with each record in the
-    // page
-    //
-    // The provider API will hopefully support pagination. Functions like this
-    // should maintain pagination state, and for each page, for each record in
-    // the page, invoke the `ResourceIteratee`. This will encourage a pattern
-    // where each resource is processed and dropped from memory.
-
-    const users: AcmeUser[] = [
-      {
-        id: 'acme-user-1',
-        name: 'User One',
-      },
-      {
-        id: 'acme-user-2',
-        name: 'User Two',
-      },
-    ];
-
-    for (const user of users) {
-      await iteratee(user);
+  public async getAccount(): Promise<Account> {
+    const uri = '/api/account/get-account';
+    const request = got.post(BASE_URI + uri, {
+      headers: this.generateHeaders(uri),
+      body: EMPTY_BODY,
+    });
+    try {
+      const result = await request;
+      const response = JSON.parse(result.body) as ApiResponse<Account>;
+      this.verifyResponse(response);
+      return response.data[0];
+    } catch (err) {
```

Comment:
going to tidy up this flow. I am only going to wrap the try/catch around the request itself and throw an error in the event it rejects. I'll let `verifyResponse` do the rest of the error handling. It will live outside the try/catch. Will also use more appropriate error types in `verifyResponse`.